### ISSUE TITLE
checkers: improve error reporting for boolExprSimplify in lambdas

### ISSUE THREAD

**Issue by Quasilyte** [go-critic/go-critic#764]
`boolExprChecker` gives false positive.
Reproducer:
```go
func returnsBool(f func()) bool { return false }

func insideLambda() {
	var x, y int

	// Code below gives a warning. But it should not.
	_ = returnsBool(func() {
		_ = !(x >= y)
	})
}
```
Linter output:
```
can simplify `returnsBool(func() {
	_ = !(x >= y)
})` to `returnsBool(func() {
	_ = x < y
})`
```
All versions of `boolExprSimlify` are affected.

**Comment by Quasilyte** [comment]
Well, it's not a false positive, but rather a too greedy way to report issues.
Instead of printing entire lambda, it should print only relevant boolean expression change.
